README.md

```diff
@@ -57,24 +136,79 @@ Apache Kudu **doesn't** support `DESCENDING` sort keys. To provide this function
 		.timestampToMicros(new Timestamp(EPOCH_FOR_REVERSE_SORT_IN_MILLISECONDS));
 ```
 
+Using these constants at write time, our system writes to Kudu a translated date field using this formula `EPOCH_FOR_REVERSE_SORT_IN_MICROSECONDS - ACTUAL-DATE-OF-EVENT`. And at read time, our system reverses this logic keeping it transparent to the end user. `Calcite-Kudu` translates timestamps at read time back into their actual values.
+This implementation allows `Calcite-Kudu` to leverage Kudu's existing sort logic eliminating the need to sort on the client.
 
-Using these constants at write time, our system writes to Kudu a translated date field using this formula `EPOCH_FOR_REVERSE_SORT_IN_MICROSECONDS - ACTUAL-DATE-OF-EVENT`. And at read time, our system reverses this logic keeping it transparent to the end user. `Calcite-Kudu` translates timestamps at read time back into their actual values. This implementation allows `Calcite-Kudu` to leverage Kudu's existing sort logic preventing costly sorting behavior in our services.
+When creating a table using DDL, specify the column as `DESC`. In the following example, the data is stored in descending order of
+`event_time`.
 
-When creating a Table using DDL, specify the column as `DESC`. In the following example, `UNIXTIME_MICROS_COL` is setup as `DESCENDING` sort.
+``` sql
+CREATE TABLE "host_metrics" (
+	"host" VARCHAR NOT NULL,
+	"metric_name" VARCHAR NOT NULL,
+	"event_time" TIMESTAMP NOT NULL ROW_TIMESTAMP,
+	"metric_value" DOUBLE NOT NULL,
+	PRIMARY KEY ("host", "metric_name", "event_time" DESC))
+PARTITION BY HASH ("host") PARTITIONS 2;
+```
+
+## Write Support
+Data can be written to kudu by using an INSERT statement, but this should only be used for testing.
+Materialized views that are defined will be automatically maintained correctly as long as the data for a day is written from a
+single client in a single session.
+``` sql
+INSERT INTO "host_metrics" VALUES ('host1', 'metric1', TIMESTAMP'2020-01-02', 10);
+```
 
+## Materialized Views
+The adapter allows the querying of rollup aggregate materialized views which can be registered using the
+`CREATE MATERIALZIED VIEW` statement. These views have to be maintained by an external system.
```

Comment:
So we are calling out that an external system(factory or something) needs to be maintained by the users creating such a materialized view ?

Reply:
The writes to the base table can be done using calcite-kudu, but the writes to the materialized view has to be done using some streaming or batch system (since the data needs to be aggregated correctly).